lib/builtIn/Macro.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const DefaultConfig = require('../DefaultConfig');
+const MACROSTEP = /^macro (.*) ?is executed$/;
+
+/**
+ * Macro to create a Cucumber step by combining a block of steps, simplifying often recurring steps.
+ * @class
+ * @extends DefaultConfig
+ */
+class Macro extends DefaultConfig {
+
+    constructor() {
+        super();
+        this.macros = {};
+    }
+
+    /**
+     * Function to filter out scenarios of a feature (scenario, scenario outline)
+     * before they are processed.
+     * Return FALSE if the given element should be filtered out. Otherwise it won't be.
+     *
+     * @param {Scenario|ScenarioOutline} scenario The scenario to be tested.
+     * @param {Feature} parent The parent feature of the scenario.
+     * @param {number} i The index of the current element.
+     * @returns {boolean|*} FALSE if the given element should be filtered out.
+     */
+    preFilterScenario(scenario, parent, i) {
+        const macroTag = scenario.tags.find(tag => /^@macro\(.*\)/.test(tag.name));
+
+        if (macroTag) {
+            const name = macroTag.name.substring(7, macroTag.name.length - 1);
+
+            if (name.length === 0) {
+                throw new Error('Name is not provided for macro for scenario ${scenario.name}.');
+            }
+
+            if (name in this.macros) {
+                throw new Error('Name ${name} already used in scenario ${this.macros[name].name}.');
+            }
+
+            if (scenario.steps.length === 0) {
+                throw new Error('Macro ${name} does not contain any steps.');
```

Comment:
@judit-nahaj I am sure, that here, you wanted to use the template literals, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals